### PR TITLE
BML: Merge the updated repo/branch with the target

### PR DESCRIPTION
**Pull request by lentzi90**
Previously we just checked out the updated branch. This always checks out the target (metal3-dev-env main) and then merges the updated repo/branch with it if there are differences.

### CODE REVIEW

**Review by kashifest — APPROVED** [review]

File: jenkins/scripts/bare_metal_lab/deploy-lab.yaml [hunk 1]
```diff
@@ -119,11 +119,24 @@
 
     - name: Clone the metal3-dev-env repo
       git:
-        repo: "{{ metal3_dev_env_repo }}"
+        repo: "https://github.com/metal3-io/metal3-dev-env.git"
         dest: "/home/{{ ansible_user_id }}/{{ metal3_dir }}"
-        version: "{{ metal3_dev_env_branch }}"
+        version: "main"
```
Comment (furkatgofurov7) [inline jenkins/scripts/bare_metal_lab/deploy-lab.yaml:124]: `metal3_dev_env_repo` and `metal3_dev_env_branch` were introduced to differentiate bml runs from this repo and m3-dev-env repo, so that if it the latter it will run the tests with updated branch. Now, what is the motivation behind this change?

**Comment by furkatgofurov7** [comment]
I am even reluctant to merge this, since BML is so unreliable nowadays :grinning:
/lgtm
